firestore/integration_test.go

```diff
@@ -1601,3 +1601,93 @@ func TestDetectProjectID(t *testing.T) {
 		t.Errorf("expected an error while using TokenSource that does not have a project ID")
 	}
 }
+
+func TestIntegration_ColGroupRefPartitions(t *testing.T) {
+	h := testHelper{t}
+	coll := integrationColl(t)
+	ctx := context.Background()
+
+	// Create a doc in the test collection so a collectionID is live for testing
+	doc := coll.NewDoc()
+	h.mustCreate(doc, integrationTestMap)
+
+	for _, tc := range []struct {
+		collectionID           string
+		expectedPartitionCount int
+	}{
+		// Verify no failures if a collection doesn't exist
+		{collectionID: "does-not-exist", expectedPartitionCount: 1},
+		// Verify a collectionID with a small number of results returns a partition
+		{collectionID: coll.collectionID, expectedPartitionCount: 1},
+		// TODO: could implement a scenario where a collectionID had a good
+		// number (200+) documents. Would require test document creation.
+		// {collectionID: "thousand", expectedPartitionCount: 4},
+	} {
+		colGroup := iClient.CollectionGroup(tc.collectionID)
+		partitions, err := colGroup.GetPartitions(ctx, 10)
+		if err != nil {
+			t.Fatalf("Did not expect error: %v", err)
+		}
+		if got, want := len(partitions), tc.expectedPartitionCount; got != want {
+			t.Errorf("Unexpected Partition Count: got %d, want %d", got, want)
+		}
+	}
+}
+
+func TestIntegration_ColGroupRefPartitionsLarge(t *testing.T) {
+	// Create collection with enough documents to have multiple partitions.
+	coll := integrationColl(t)
+	collectionID := coll.collectionID + "largeCollection"
+	coll = iClient.Collection(collectionID)
+
+	ctx := context.Background()
+
+	documentCount := 2*128 + 127 // Minimum partition size is 128.
+	expectedPartitionCount := 4
+
+	// Create documents in a collection sufficient to trigger multiple partitions.
+	batch := iClient.Batch()
+	deleteBatch := iClient.Batch()
+	for i := 0; i < documentCount; i++ {
+		doc := coll.Doc(fmt.Sprintf("doc%d", i))
+		batch.Create(doc, integrationTestMap)
+		deleteBatch.Delete(doc)
+	}
+	batch.Commit(ctx)
+	defer deleteBatch.Commit(ctx)
+
+	// Verify that we retrieve 383 documents for the colGroup (128*2 + 127)
+	colGroup := iClient.CollectionGroup(collectionID)
+	docs, err := colGroup.Documents(ctx).GetAll()
+	if err != nil {
+		t.Fatalf("Did not expect error: %v", err)
+	}
+	if got, want := len(docs), documentCount; got != want {
+		t.Errorf("Unexpected number of documents in collection group: got %d, want %d", got, want)
+	}
+
+	// Get partitions, allow up to 10 to come back, expect less will be returned.
+	partitions, err := colGroup.GetPartitions(ctx, 10)
```

Comment:
Our other tests don't seem to validate the actual partition count, though I'd expect more than one partition. I could soften this to that. With 383 docs, here is what the partitions are:
```
Start:
  End: proj/.../(default)/documents/largeCollection/doc121

Start: proj/.../(default)/documents/largeCollection/doc121
  End: proj/.../(default)/documents/largeCollection/doc232

Start: proj/.../(default)/documents/largeCollection/doc232
  End: proj/.../(default)/documents/largeCollection/doc34

Start: proj/.../(default)/documents/largeCollection/doc34
  End:
```